optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/Sequence.java

```diff
@@ -41,10 +45,32 @@ protected Sequence(ConsecutiveSetTree<ValueType_, ?, ?> sourceTree, NavigableSet
         this.count = count;
     }
 
-    public NavigableSet<ValueType_> getItems() {
+    public Collection<ValueType_> getItems() {
         return consecutiveItemsSet;
     }
 
+    public ValueType_ getFirstItem() {
+        if (firstItem == null) {
+            firstItem = consecutiveItemsSet.first();
+        }
+        return firstItem;
+    }
+
+    public ValueType_ getLastItem() {
+        if (lastItem == null) {
+            lastItem = consecutiveItemsSet.last();
+        }
+        return lastItem;
+    }
+
+    public ValueType_ getItemBefore(ValueType_ item) {
+        return consecutiveItemsSet.lower(item);
+    }
+
+    public ValueType_ getItemAfter(ValueType_ item) {
+        return consecutiveItemsSet.higher(item);
+    }
```

Comment:
We could possibly live without these convenience methods.
That said, they allow me to make `getItems()` a `Collection` and clearly state that it should only be used for iteration.